src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java

```diff
@@ -69,6 +72,11 @@ public void onEnable() {
 
         setupResearches();
         new ThoriumResource().register();
+
+        if (WaterSprinkler.successChance.getValue() < 0 || WaterSprinkler.successChance.getValue() > 1) {
+            getLogger().warning("Water Sprinkler success chance has to be between 0-1, resetting to default (0.5)");
+            WaterSprinkler.successChance.update(0.5);
+        }
```

Comment:
Override the `validate` method for your ItemSetting or create a new subclass, that is exactly what this is for.